source/frontend/src/components/common/mapFSM/machineDefinition/mapMachine.ts

```diff
@@ -0,0 +1,299 @@
+import { latLngBounds } from 'leaflet';
+import { assign, createMachine, raise } from 'xstate';
+
+import { defaultBounds } from '@/components/maps/constants';
+import { defaultPropertyFilter } from '@/features/properties/filter/IPropertyFilter';
+
+import { emptyFeatureData } from '../models';
+import { MachineContext, SideBarType } from './types';
+
+const featureDataStates = {
+  initial: 'idle',
+  states: {
+    idle: {
+      on: {
+        REFRESH_PROPERTIES: {
+          actions: assign({ isLoading: () => true }),
+          target: 'loading',
+        },
+        SET_MAP_FILTER: {
+          actions: assign({ isLoading: () => true, mapFilter: (_, event: any) => event.mapFilter }),
+          target: 'loading',
+        },
+      },
+    },
+    loading: {
+      invoke: {
+        src: 'loadFeatures',
+        onDone: {
+          target: 'idle',
+          actions: [
+            assign({
+              isLoading: () => false,
+              mapFeatureData: (_, event: any) => event.data,
+            }),
+            raise('REQUEST_FIT_BOUNDS'),
+          ],
+        },
+      },
+    },
+  },
+};
+
+const mapRequestStates = {
+  initial: 'nothingPending',
+  states: {
+    nothingPending: {
+      on: {
+        REQUEST_FLY_TO_LOCATION: {
+          actions: assign({
+            requestedFlyTo: (_, event: any) => ({
+              bounds: null,
+              location: event.latlng,
+            }),
+          }),
+          target: 'pendingFlyTo',
+        },
+        REQUEST_FLY_TO_BOUNDS: {
+          actions: assign({
+            requestedFlyTo: (_, event: any) => ({
+              bounds: event.bounds,
+              location: null,
+            }),
+          }),
+          target: 'pendingFlyTo',
+        },
+        REQUEST_FIT_BOUNDS: {
+          actions: assign((context: any) => {
+            if (context.draftLocations.length > 0) {
```

Comment:
I could be missing something but it seems like "draft" properties is being used for both properties to highlight when selecting and properties to highlight when viewing a file. I'd argue for at least a rename, draft to me implies these properties have not been saved.

Reply:
My intention was to get feature parity, so having it named the same would help match the functionality. I do believe that renaming is better.